kratos/modified_shape_functions/triangle_2d_3_modified_shape_functions.cpp

```diff
@@ -0,0 +1,190 @@
+void Triangle2D3ModifiedShapeFunctions::PrintData(std::ostream& rOStream) const {
+    const GeometryPointerType p_geometry = this->GetInputGeometry();
+    const Vector nodal_distances = this->GetNodalDistances();
+    rOStream << "Triangle2D3N modified shape functions computation class:\n";
+    rOStream << "\tGeometry type: " << (*p_geometry).Info() << "\n";
+    std::stringstream distances_buffer;
+    for (unsigned int i = 0; i < nodal_distances.size(); ++i) {
+        distances_buffer << std::to_string(nodal_distances(i)) << " ";
+    }
+    rOStream << "\tDistance values: " << distances_buffer.str();
+};
+
+// Returns true if the element is splitting
+bool Triangle2D3ModifiedShapeFunctions::IsSplit() {
+    return mpTriangleSplitter->mIsSplit;
+};
+
+// Internally computes the splitting pattern and returns all the shape function values for the positive side.
+void Triangle2D3ModifiedShapeFunctions::GetPositiveSideShapeFunctionsAndGradientsValues(
+    Matrix &rPositiveSideShapeFunctionsValues,
+    std::vector<Matrix> &rPositiveSideShapeFunctionsGradientsValues,
+    Vector &rPositiveSideWeightsValues,
+    const IntegrationMethodType IntegrationMethod) {
+
+    if (this->IsSplit()) {
+        // Get the intersection points condensation matrix
+        Matrix p_matrix;
+        SetIntersectionPointsCondensationMatrix(p_matrix,
+                                                mpTriangleSplitter->mEdgeNodeI,
+                                                mpTriangleSplitter->mEdgeNodeJ,
+                                                mpTriangleSplitter->mSplitEdges,
+                                                mpTriangleSplitter->mSplitEdgesSize);
+
+        // Compute the positive side values
+        this->ComputeValuesOnOneSide(rPositiveSideShapeFunctionsValues,
+                                        rPositiveSideShapeFunctionsGradientsValues,
+                                        rPositiveSideWeightsValues,
+                                        mpTriangleSplitter->mPositiveSubdivisions,
+                                        p_matrix,
+                                        IntegrationMethod);
+    } else {
+        KRATOS_ERROR << "Using the GetPositiveSideShapeFunctionsAndGradientsValues method for a non divided geometry.";
+    }
+};
+
+// Internally computes the splitting pattern and returns all the shape function values for the negative side.
+void Triangle2D3ModifiedShapeFunctions::GetNegativeSideShapeFunctionsAndGradientsValues(
+    Matrix &rNegativeSideShapeFunctionsValues,
+    std::vector<Matrix> &rNegativeSideShapeFunctionsGradientsValues,
+    Vector &rNegativeSideWeightsValues,
+    const IntegrationMethodType IntegrationMethod) {
+
+    if (this->IsSplit()) {
+        // Get the intersection points condensation matrix
+        Matrix p_matrix;
+        SetIntersectionPointsCondensationMatrix(p_matrix,
+                                                mpTriangleSplitter->mEdgeNodeI,
```

Comment:
note that if you want to get a c-array from a vector, you can use the ".data()" method

Reply:
Done.